### DISCUSSION TITLE
Having issues with PyAEDT installation for 22R2

### DISCUSSION THREAD

**Discussion by Abhi-SimuTech** [ansys/pyaedt#4741] [Q&A] [answered]
Hi Everyone,

I am having some issue with PyAEDT installation for 22R2 on my laptop with Windows 10. Please see the error message below. 
PyAEDT used to work just fine with 22R2 before on the same machine. However, I uninstalled 22R2 earlier this year, and now when I re-installed 22R2 I am not being able to install PyAEDT anymore. 

Kindly suggest a possible fix. Thanks.

P.S. I am have been trying to use the "PyAEDT Installer Python" file and tried to install for 22R2 w/ or w/o the update patch. Issue is the same.

![image](https://github.com/ansys/pyaedt/assets/145813888/c623ab4e-48eb-4651-8b0b-e30bfa648245)

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @Abhi-SimuTech ,

You are right, this week a new module was introduced, that is not compatible with Python 3.7, which is the python version used by 2022R2 and 2023R1. But, we are lucky, and yesterday it was removed from the dependencies:
https://github.com/ansys/pyaedt/commit/196a87c2bc4c5252364a306e7230ace8806b893a#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711

It means, in the next PyAEDT version 0.9.3, it will work again. This new release will happen today or tomorrow. I will let you know.

In the meanwhile, you can test it, by commenting and uncommenting these lines:

![image](https://github.com/ansys/pyaedt/assets/85613111/923ee75d-34de-4b6b-91a0-c1f1c6bb86f5)


This will install PyAEDT from the main, instead of using the latest release. You need to have git installed on your computer. For me it worked.

Finally, I suggest moving to new AEDT versions, because 2022R2 is the last PyAEDT-compatible version and for example, in some months it will not be compatible with new PyAEDT versions, of course, if you want to use PyAEDT 0.9.3 and AEDT 2022R2, you can use it forever, but if you need to update PyAEDT versions ( for instance PyAEDT v1.0, at some point in the future, this AEDT version will not be compatible. 

Right now, to have better experience, I suggest using AEDT 2024R1, because all PyAEDT unit tests are passing on this version and it is using Python 3.10 by default which is the standard in the community. Python 3.7 is deprecated, other modules like matplotlib or numpy, are already stopped, and it means if there is a bug in Python 3.7 related to matplotlib, they will not fix it, for that reason, it is better to move to AEDT 2024R1.

**Comment by Samuelopez-ansys** [comment]
@Abhi-SimuTech Did you try with the new installer and the PyAEDT version?
